packages/xchain-doge/__tests__/client.test.ts

```diff
@@ -0,0 +1,317 @@
+    expect(() => dogeClient.setPhrase('cat')).toThrow()
+  })
+
+  it('should not throw an error for setting a good phrase', () => {
+    expect(dogeClient.setPhrase(phraseOne)).toBeUndefined
+  })
+
+  it('should purge phrase and utxos', async () => {
+    dogeClient.purgeClient()
+    expect(() => dogeClient.getAddress()).toThrow('Phrase must be provided')
+  })
+
+  it('should validate the right address', () => {
+    dogeClient.setNetwork('testnet' as Network)
+    dogeClient.setPhrase(phraseOne)
+    const address = dogeClient.getAddress()
+    const valid = dogeClient.validateAddress(address)
+    expect(address).toEqual('nfVtm2hbz4rv1fhBVTqi5hGWrRzy5rqmLv')
+    expect(valid).toBeTruthy()
+  })
+
+  it('should invalidate the address', () => {
+    dogeClient.setNetwork('testnet' as Network)
+    dogeClient.setPhrase(phraseOne)
+    const invalid1 = dogeClient.validateAddress('NFVTM2HBZ4RV1FHBVTQI5HGWRRZY5RQMLV')
+    const invalid2 = dogeClient.validateAddress('nfvtm2hbz4rv1fhbvtqi5hgwrrzy5rqmlv')
+    expect(invalid1).toBeFalsy()
+    expect(invalid2).toBeFalsy()
+  })
+
+  it('set phrase should return correct address', () => {
+    dogeClient.setNetwork('testnet' as Network)
+    expect(dogeClient.setPhrase(phraseOne)).toEqual(testnet_address_path0)
+    expect(dogeClient.getAddress(1)).toEqual(testnet_address_path1)
+
+    dogeClient.setNetwork('mainnet' as Network)
+    expect(dogeClient.setPhrase(phraseOne)).toEqual(mainnet_address_path0)
+    expect(dogeClient.getAddress(1)).toEqual(mainnet_address_path1)
+  })
+
+  it('should get the right balance', async () => {
+    const expectedBalance = 10000000000
+    dogeClient.setNetwork('testnet' as Network)
+    dogeClient.setPhrase(phraseOne)
+    const balance = await dogeClient.getBalance(dogeClient.getAddress())
+    expect(balance.length).toEqual(1)
+    expect(balance[0].amount.amount().toNumber()).toEqual(expectedBalance)
+  })
+
+  it('should get the balance of an address without phrase', async () => {
+    dogeClient.setNetwork('testnet' as Network)
+    dogeClient.purgeClient()
+    const balance = await dogeClient.getBalance('nkJ3JHmpuWqu3eHyXByh67YmwDGep9bD3N')
+    expect(balance.length).toEqual(1)
+    expect(balance[0].amount.amount().toNumber()).toEqual(1739730667)
+  })
+
+  it('should get the right balance when scanUTXOs is called twice', async () => {
+    const expectedBalance = 10000000000
+    dogeClient.setNetwork('testnet' as Network)
+    dogeClient.setPhrase(phraseOne)
+
+    const balance = await dogeClient.getBalance(dogeClient.getAddress())
+    expect(balance.length).toEqual(1)
+    expect(balance[0].amount.amount().toNumber()).toEqual(expectedBalance)
+
+    const newBalance = await dogeClient.getBalance(dogeClient.getAddress())
+    expect(newBalance.length).toEqual(1)
+    expect(newBalance[0].amount.amount().toNumber()).toEqual(expectedBalance)
+  })
+
+  // Some random 404 and useless error?
```

Comment:
Couldn't get these tests working?

Reply:
We might need default implementation here for stagenet for these tests to work.

Reply:
@towanTG Could you check this, please?

Reply:
Hey guys, was ill the last couple of days. fixing the issues now. you merged already though. should I create a new pull request for the tests? @pluto-9r @polaris-9r

Reply:
> Hey guys, was ill the last couple of days. fixing the issues now. you merged already though. should I create a new pull request for the tests? @pluto-9r @polaris-9r

@towanTG Yes, pls open another PR. Since last issues are not critical, I've just merged + published this PR to let other wallet devs to start their integration with `xchain-doge`.